packages/rpc/src/WsClient.ts

```diff
@@ -100,44 +95,54 @@ export default class WsClient extends Client {
   }
 
   protected async send<const T extends RpcMethod>(
-    data: JsonRpcRequest<T>,
-  ): Promise<{ success: true; result: unknown } | { success: false; error: Error }> {
-    let requestId = data.id;
-
-    for (let i = 0; i < 5; i += 1) {
-      let socket;
-      try {
-        socket = await this.connectionReady();
-      } catch {
-        // retry
-        continue;
+    requests: JsonRpcRequest<T>[],
+  ): Promise<Response[]> {
+    const responses: Response[] = [];
+    for (const data of requests) {
+      const MAX_RETRIES = 5;
+      for (let i = 0; i < MAX_RETRIES; i += 1) {
+        let socket;
+        try {
+          socket = await this.connectionReady();
+        } catch {
+          // retry
+          continue;
+        }
+
+        socket.send(JSON.stringify(data));
```

Comment:
can we batch these too?
```suggestion
        socket.send(JSON.stringify(requests));
```

Reply:
we can but I think it's kinda pointless. WS is supposed to handle this case without extra optimisation

Reply:
added a new method for sending batch to the HttpClient and the old method would work with single requests for now

Reply:
dont understand the last comment here - which one is the new method?

Reply:
sorry it's not a new method, but an override of `sendRequest` on the `HttpClient`.

Reply:
so the original method would work with single requests which is being used by the WebSocket class.